.circleci/extended/s3-publish.yml

```diff
@@ -0,0 +1,48 @@
+# This config uses the `.circleci/extended/orb-test.yml` as its base, and then publishes the app to S3.
+# The workflow contained here demonstrates a practical application of `post-steps`
+# and using workspace persistence
+version: 2.1
+
+orbs:
+  node: circleci/node@4.1
+  # The S3 orb will setup the aws-cli and quickly copy or sync anything to s3.
+  # Orbs reduce the amount of configuration required for common tasks.
+  # See the orb documentation here: https://circleci.com/developer/orbs/orb/circleci/aws-s3
+  # NOTE: Environment variables containing the necessary secrets can be setup in the CircleCI UI
+  # See here https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-project
+  aws-s3: circleci/aws-s3@2.0
+
+jobs:
+  s3-publish:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - attach_workspace:
+            # this adds the dist directory to the workspace, which by default is /home/circleci/project/
```

Comment:
@tsully - I was just realizing - I could actually just make this config look like this ... Which doesn't really demo the workspace attachment... Maybe there needs to be a better use case for the persist to workspace as an example? Thoughts?

```yaml

version: 2.1

orbs:
  node: circleci/node@4.1
  aws-s3: circleci/aws-s3@2.0

workflows:
  sample:
    jobs:
      - node/test:
          version: "15.1"
          # after the node/test job completes, build the dist packages and save them to a workspace
          post-steps:
            - run: npm run build
            - aws-s3/copy:
                from: dist/
                to: s3://sample-app-vue
                arguments: --recursive


```

Reply:
Nice! I like that this is simpler and I don't know if it's necessary to demo the workspace attachment feature. The only reason I'd prefer the previous config is if it's very likely that users deploying to S3 would add on an additional job that would require the use of workspace. Otherwise, the new config works well.

Reply:
hmm... possibly, but I think we could have a better example for that with e2e testing or something

Reply:
Demoing workspaces isnt as important as showing useful real-world examples imo.
For workspace demos, It might make sense to build the android app for this project and persist that to an android test.